mantidproject/mantid
# Make Improvements to the Engineering Diffraction 2 GUI

### ~Blocked By: #27486 #27589~

**Description of work.**
A few changes have been made to the code and functionality of the engineering diffraction GUI. 

-  Vanadium run files are also saved to the user directories.
- Focus outputs now include `.gss`, `.dat`, and `.nxs` files.
- The user is provided with more status information, such as when the GUI is searching for files.
- Similar plots for multiple banks are shown as tiled plots.
- Refactored current calibration into separate class.
- Added Documentation for focusing tab
- Made Ceria/Calib variables consistent, now using "`sample`"

**To test:**
1. Open Eng Diff GUI.
2. Click the calibrate button while the GUI is searching for files. 
2. Create a new calibration with "Plot Output" ticked (V: 307521 Sample: 305738)
3. Check similar Plots appear in the same figure. 
1. Check documentation is up to date. 
0. Go to the focus tab and focus a run (305761)
11. Check that all 3 file types have been generated. 

Fixes #27189 

*This does not require release notes* because **users have not had access to this GUI yet.**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
